Sources/Tentacle/Client.swift

```diff
@@ -99,12 +94,161 @@ extension URLSession {
 	}
 }
 
-enum HTTPMethod: String {
-    case get = "GET"
-    case post = "POST"
-    case put = "PUT"
-    case head = "HEAD"
-    case options = "OPTIONS"
+internal struct Request {
+    enum Method: String {
+        case get = "GET"
+        case post = "POST"
+        case put = "PUT"
+        case head = "HEAD"
+        case options = "OPTIONS"
+    }
+    
+    var method: Method
```

Comment:
Not really.

I try to default to `var` with `struct`s, using `let` only when it would create an invalid value, since `let`ness applies recursively.